cmd/main/main.go

```diff
@@ -357,15 +358,17 @@ func main() {
 	}
 
 	privateHTTPServer := &http.Server{
-		Addr:      fmt.Sprintf(":%v", config.Config.Server.PrivatePort),
-		Handler:   grpcHandlerFunc(privateGrpcS, privateServeMux),
-		TLSConfig: tlsConfig,
+		Addr:              fmt.Sprintf(":%v", config.Config.Server.PrivatePort),
+		Handler:           grpcHandlerFunc(privateGrpcS, privateServeMux),
+		TLSConfig:         tlsConfig,
+		ReadHeaderTimeout: 10 * time.Millisecond,
```

Comment:
This is a bit aggressive (although for headers should be fine). What did you base your choice on?

Reply:
Security issue (https://cwe.mitre.org/data/definitions/400.html)
If we don't setup the header timeout, what if we have million or billion request come in with malice, it would consume all of our service resource.

Reply:
Yeah, I meant more the actual value (10ms vs, for instance, 100ms). I agree it's better to cap the header timeout.

Reply:
Ah, I missed. The HTTP header parser usually wouldn't take over 1 ms (for 1KB size). So 5-10 ms is enough. Unless our HTTP header over 1KB, maybe we need to set more longer.